### PR TITLE
Log errors

### PR DESCRIPTION
**Pull request by RussKie**
An attempt to capture an elusive error information for #9166.
Write an error log before showing the bug reporter dialog if `AppSettings.WriteErrorLog=true`.

Contributes to #9166

### CODE REVIEW

**Review by gerhardol — APPROVED** [review]
The description in #9166 should be added to the description

File: GitUI/NBugReports/BugReportInvoker.cs [hunk 1]
```diff
@@ -68,8 +68,30 @@ void AppendIfNotEmpty(string? value, string designation)
...
+            string tempFile = Path.Combine(tempFolder, tempFileName);
+
+            try
+            {
+                string content = $"Is fatal: {isTerminating}\r\n{exception}";
+                File.WriteAllText(tempFile, content);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show($"Failed to error to {tempFile}\r\n{ex.Message}", "Error writing log", MessageBoxButtons.OK, MessageBoxIcon.Error);
```
Comment (mstv) [inline GitUI/NBugReports/BugReportInvoker.cs:84]:
```suggestion
                MessageBox.Show($"Failed to log error to {tempFile}\r\n{ex.Message}", "Error writing log", MessageBoxButtons.OK, MessageBoxIcon.Error);
```

File: GitUI/NBugReports/BugReportInvoker.cs [hunk 2]
```diff
@@ -68,8 +68,30 @@ void AppendIfNotEmpty(string? value, string designation)
...
         }
 
+        private static void LogError(Exception exception, bool isTerminating)
+        {
+            string tempFolder = Path.GetTempPath();
+            string tempFileName = $"{AppSettings.ApplicationId}.{AppSettings.AppVersion}.{DateTime.Now.ToString("yyyyMMdd.HHmmssfff")}.log";
+            string tempFile = Path.Combine(tempFolder, tempFileName);
+
+            try
+            {
```
Comment (mstv) [inline GitUI/NBugReports/BugReportInvoker.cs:78]: I am getting paranoid:
```suggestion
            try
            {
                string tempFolder = Path.GetTempPath();
                string tempFileName = $"{AppSettings.ApplicationId}.{AppSettings.AppVersion}.{DateTime.Now.ToString("yyyyMMdd.HHmmssfff")}.log";
                string tempFile = Path.Combine(tempFolder, tempFileName);

```

**Review by mstv — APPROVED** [review]
Two nits

File: GitUI/NBugReports/BugReportInvoker.cs (hunk 2 above)
Comment (RussKie) [reply GitUI/NBugReports/BugReportInvoker.cs:78]: We can't do that - `tempFile` is used to form the exception message.

**Pull request merged**